tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+    '''
+    gold_path is the path to the directory containing the gold standard files
+    raw_path is the path to the directory containing the raw files
+    '''
+    n_processed = {} 
+    raw_files = os.listdir(raw_path)
+    n_files = len(raw_files)
+    eval_list = {}
+    sum = {}
+    ss = {}
+    for library in list_of_libraries:
+        sum[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ss[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        n_processed[library.__name__] = 0
+    for filename in raw_files:
+        raw_file = raw_path + '/' + filename
+        gold_standard_file = gold_path + '/' + filename
+        if not os.access(gold_standard_file, os.R_OK):
+            continue  # original and cleaned files don't always match in CleanEval gold standard
+        raw_html = slurp_file(raw_file)
+        gold_html = normalize(slurp_file(gold_standard_file), opt_ascii, opt_unlabelled)
+        for library in list_of_libraries:
+            cleaned_html = library(raw_html)
+            eval_list[library.__name__] = eval_method(gold_html, cleaned_html)
+            if not opt_total:
+                print(raw_file+ " ("+library.__name__+")" + "\t" + ("%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%d\t%d\t%d\t%d\t%d\t%d" % eval_list[library.__name__]))
+            for i, x in enumerate(eval_list[library.__name__]):
+                sum[library.__name__][i] += x
+                ss[library.__name__][i] += x ** 2
+            n_processed[library.__name__] += 1
+    for library in list_of_libraries:
+        if n_processed[library.__name__] < n_files:
+            print("Warning: %d files skipped for %s" % (n_files - n_processed[library.__name__], library.__name__))
+    if not opt_latex:
+        print("Library\t\t\tF\tP\tR\tF.tag\tP.tag\tR.tag\tTP\tFP\tFN\tTP.tag\tFP.tag\tFN.tag")
+    else:
+        print(r'\begin{tabular}{ |c|c|c|c|c|c|c|c|c|c|c|c|c| }')
+        print(r'\hline')
+        print(r'Library & F & P & R & F.tag & P.tag & R.tag & TP & FP & FN & TP.tag & FP.tag & FN.tag \\')
+        print(r'\hline')
+    for library in list_of_libraries:
+        tp = sum[library.__name__][6]
+        fp = sum[library.__name__][7]
+        fn = sum[library.__name__][8]
+        tag_tp = sum[library.__name__][9]
+        tag_fp = sum[library.__name__][10]
+        tag_fn = sum[library.__name__][11]
+        result = __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn)
+        if not opt_latex:
+            print(library.__name__ + "\t\t" + ("%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%d\t%d\t%d\t%d\t%d\t%d\n" % result))
+        else:
+            print(r'%s & ' % library.__name__ + r'%.2f & %.2f & %.2f & %.2f & %.2f & %.2f & %d & %d & %d & %d & %d & %d \\' % result)
+            print(r'\hline')
+
+
+cache_dir = "tests/.cache/"
+gold_dir = "tests/.golden/"
+
+def load_data():
```

Comment:
this is a very generic function name with no docstring; rename the function to specify what data is being loaded, and add a docstring

Reply:
most of your functions are large blobs of code with no line breaks and no explanations about why you chose write the code the way you did

you should modify this so that the code is divided into "paragraphs" separated by a line break and with a single comment at the top of the paragraph that explains what/why the code is doing; for examples, see the code I wrote in the `tests/test_content.py` file: https://github.com/mikeizbicki/metahtml/blob/f60ece260fbb0b33e23cb2515b406cd50c58fd9c/tests/test_content.py#L195

This is formatting is particularly nice in vim; `{` and `}` jumps the cursor between paragraphs (and `[`/`]` jumps the cursor between function definitions), and so code is really easy to navigate when formatted this way